tidy3d/components/simulation.py

```diff
@@ -401,7 +401,9 @@ def _warn_monitor_simulation_frequency_range(cls, val, values):
             return val
 
         # Get simulation frequency range
-        source_ranges = [source.source_time.frequency_range() for source in values.get("sources")]
+        if "sources" not in values:
+            return val
```

Comment:
let's instead 
```py
raise ValidationError("could not validate `_warn_monitor_simulation_frequency_range` as `sources` failed validation")
```
here.  if `"sources" not in values:`, it means something went wrong and we want to make sure that we raise an error instead of just returning `val`, which indicates that everything is fine.